crates/nixci/src/nix/ssh.rs

```diff
@@ -0,0 +1,67 @@
+//! Function for working with `nix run on ssh`.
+use std::path::{Path, PathBuf};
+
+use crate::{config::ref_::ConfigRef, step::build::BuildStepArgs};
+
+use anyhow::Context;
+use tokio::process::Command;
+
+/// Runs `nix run command through ssh on remote machine` in Rust
+///
+pub async fn on_ssh(
+    build_step_args: &BuildStepArgs,
+    remote_address: &str,
+    omnix_input: &Path,
+    flake_url: PathBuf,
+    cfg_ref: ConfigRef,
+) -> anyhow::Result<()> {
```

Comment:
@rsrohitsingh682 I think you need to rethink this function entirely. 

First of all, anything under the `nix/` package should *not* be depending on the rest of the `nixci` crate. Which is kind of obvious.  The `nix/` package contains Nix-specific stuff; not nixci specific stuff. The use of both `BuildStepArgs` and `ConfigRef` makes this otherwise general interface of the module **complect** with the rest of nixci. See https://www.infoq.com/presentations/Simple-Made-Easy/

For that similar reason, this module should not of course be aware of the application that uses it; i.e, Omnix. Ergo, there should not be an argument called `omnix_input`. 

You should also document the function in full, and explain what each argument does.

Reply:
Naturally, the body of the function too complects, thus violating simplicity principle (per Rich Hickey presentation above).